rust/cloud-storage/search_processing_service/src/process/email/upsert.rs

```diff
@@ -23,7 +23,13 @@ pub async fn process_upsert_message(
         return Ok(());
     };
 
-    let updated_at = EpochSeconds::new(Utc::now().timestamp())?;
+    let now = EpochSeconds::new(Utc::now().timestamp())?;
+
+    let updated_at = message_info
+        .internal_date_ts
+        .map(|date| EpochSeconds::new(date.timestamp()))
+        .transpose()?
+        .unwrap_or(now);
```

Comment:
This code is repeated, could we dry it up? Also should we be erroring if the time doesn't exist? idk if it makes sense to insert now